src/coupling/assembly_base.hh

```diff
@@ -224,77 +399,90 @@ public:
 	}
 
     /// Register cell points of volume integral
-    inline void add_patch_bulk_points(const RevertableList<BulkIntegralData> &bulk_integral_data) override {
-        for (unsigned int i=0; i<bulk_integral_data.permanent_size(); ++i) {
-            if (bulk_integral_data[i].cell.dim() != dim) continue;
-            uint element_patch_idx = this->element_cache_map_->position_in_cache(bulk_integral_data[i].cell.elm_idx());
-            uint elm_pos = fe_values_->register_element(bulk_integral_data[i].cell, element_patch_idx);
-            uint i_point = 0;
-            for (auto p : this->bulk_points(element_patch_idx) ) {
-                fe_values_->register_bulk_point(bulk_integral_data[i].cell, elm_pos, p.value_cache_idx(), i_point++);
+    inline void add_patch_bulk_points() override {
```

Comment:
Move these methods into PatchFeValues, possibly wrap revertible lists into auxiliary class and pass them all
to a single  top level patch fe values update method. This simplifies call from Generic Assembly::patch_reinit.